proposals/0051-serialized-manifest.md

```diff
@@ -0,0 +1,142 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+**If you have feedback and the feature is released as experimental, please leave it on the Stage 3 PR. Otherwise, comment on the Stage 2 issue (links below).**
+
+- Start Date: 2025-01-22
+- Reference Issues: 
+- Implementation PR: https://github.com/withastro/astro/pull/13084
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/1099
+- Stage 3 PR: https://github.com/withastro/roadmap/pull/1106
+
+# Summary
+
+Expose common Astro config properties to users and integrations
+
+# Example
+
+```js
+import { trailingSlash, i18n, build } from 'astro:config/client';
+
+console.log(trailingSlash);
+console.log(i18n.locales);
+console.log(build.format);
+```
+
+```js
+import { srcDir, build } from 'astro:config/server';
+
+console.log(srcDir);
+console.log(build.client)
+console.log(build.server)
+```
+
+# Background & Motivation
+
+It happens quite often in userland to need the trailingSlash setting, or the base etc. Some things are available at import.meta.env and some that are not. The goal is to unify the way we expose this data.
+
+Many integrations need this data and have to create virtual modules for this anyway.
+
+# Goals
+
+- Have a unified way of exposing Astro config data.
+- Expose information that could be directly by our users without the use of integrations.
+- Expose information that can be used in client-side scripts.
+
+# Non-Goals
+
+- Expose non-serializable information such as integrations (they are functions).
+- Expose all information coming from the configuration.
+- Deprecations of duplicated information, which can be decided in a later stage.
```

Comment:
I understand the reason behind this but I also wish we managed to agree on a strategy. I'm afraid if we forget and keep too many ways of doing the same thing. I think it would be great to at least deprecate `import.meta.env` stuff

Reply:
I think it's fine to defer the decision on deprecation. Deprecating something at the same time as introducing its replacement can give a bit of a whiplash feeling.

Reply:
@florian-lefebvre we can use this topic in one of our TBD: plan it, discuss it with people, maybe create a thread with maintainers and collect wider feedback. Would that resolve some of your concerns?

Reply:
Yes :+1: